csrc/logical_domain_map.cpp

```diff
@@ -325,6 +325,27 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseLogicalDomainMap::map(
     return dom_map;
   }
 
+  if (IndexAccumulateOp* iaop =
+          dynamic_cast<IndexAccumulateOp*>(consumer_tv_->definition())) {
+    // Producers:
+    //     accumulate [ vocab, hidden ]
+    //     index [ * ]
+    //     value [ *, hidden ]
+    // Consumers:
+    //     output [ vocab, hidden ]
+    auto ndims_out = consumer_root.size();
+    if (producer_tv_->sameAs(iaop->accumulateTv())) {
+      for (auto idx : c10::irange(ndims_out)) {
+        updatePairwiseLogicalDomainMap(
+            producer_logical.at(idx), consumer_root.at(idx));
+      }
+    } else if (producer_tv_->sameAs(iaop->valueTv())) {
+      updatePairwiseLogicalDomainMap(
+          producer_logical.at(ndims_out - 1), consumer_root.at(ndims_out - 1));
```

Comment:
Value can have different number of dimensions than output from the comment above. But we are using the same `ndims` for them. That sounds incorrect.

Reply:
Good catch. Let me fix that.

I'll also try to figure out how to put a test for that.

Reply:
Updated the logic to remove the special code path (per Naoya's suggestion).